node-commons/src/main/java/org/apache/iotdb/commons/utils/PathUtils.java

```diff
@@ -48,118 +49,131 @@ public static String[] splitPathToDetachedNodes(String path) throws IllegalPathE
-      if (measurement.startsWith(TsFileConstant.BACK_QUOTE_STRING)
-          && measurement.endsWith(TsFileConstant.BACK_QUOTE_STRING)) {
-        if (checkBackQuotes(measurement.substring(1, measurement.length() - 1))) {
-          continue;
-        } else {
-          throw new IllegalPathException(measurement);
-        }
-      }
-      if (IoTDBConstant.reservedWords.contains(measurement.toUpperCase())
-          || isRealNumber(measurement)
-          || !TsFileConstant.NODE_NAME_PATTERN.matcher(measurement).matches()) {
-        throw new IllegalPathException(measurement);
-      }
+      res.add(checkAndReturnSingleMeasurement(measurement));
     }
+    return res;
   }
 
   /**
    * check whether measurement is legal according to syntax convention measurement could be like a.b
    * (more than one node name), in template?
    */
-  public static void isLegalMeasurementLists(List<List<String>> measurementLists)
-      throws IllegalPathException {
+  public static List<List<String>> checkIsLegalMeasurementListsAndUpdate(
+      List<List<String>> measurementLists) throws IllegalPathException {
     if (measurementLists == null) {
-      return;
+      return null;
     }
+    List<List<String>> res = new ArrayList<>();
     for (List<String> measurementList : measurementLists) {
-      isLegalMeasurements(measurementList);
+      res.add(checkIsLegalMeasurementsAndUpdate(measurementList));
     }
+    return res;
   }
 
   /**
    * check whether measurement is legal according to syntax convention measurement could be like a.b
    * (more than one node name), in template?
    */
-  public static void isLegalMeasurements(List<String> measurements) throws IllegalPathException {
+  public static List<String> checkIsLegalMeasurementsAndUpdate(List<String> measurements)
+      throws IllegalPathException {
     if (measurements == null) {
-      return;
+      return null;
     }
+    List<String> res = new ArrayList<>();
     for (String measurement : measurements) {
       if (measurement != null) {
-        PathUtils.isLegalPath(measurement);
+        res.add(PathUtils.isLegalPath(measurement)[0]);
       }
     }
+    return res;
+  }
+
+  /** check a measurement and update it if needed to. for example: `sd` -> sd */
+  public static String checkAndReturnSingleMeasurement(String measurement)
+      throws IllegalPathException {
+    if (measurement == null) {
+      return null;
+    }
+    if (measurement.startsWith(TsFileConstant.BACK_QUOTE_STRING)
+        && measurement.endsWith(TsFileConstant.BACK_QUOTE_STRING)) {
+      if (checkBackQuotes(measurement.substring(1, measurement.length() - 1))) {
+        return removeBackQuotesIfNecessary(measurement);
+      } else {
+        throw new IllegalPathException(measurement);
+      }
+    }
+    if (IoTDBConstant.reservedWords.contains(measurement.toUpperCase())
+        || isRealNumber(measurement)
+        || !TsFileConstant.NODE_NAME_PATTERN.matcher(measurement).matches()) {
```

Comment:
what's the difference with `!TsFileConstant.IDENTIFIER_PATTERN.matcher(unWrapped).matches()`